client/client.go

```diff
@@ -217,6 +217,40 @@ func (c *LogClient) UpdateRoot(ctx context.Context) error {
 	return nil
 }
 
+// VerifyInclusion ensures that the given leaf data has been included in the log.
+func (c *LogClient) VerifyInclusion(ctx context.Context, data []byte) error {
+	leaf := c.buildLeaf(data)
+	if err := c.UpdateRoot(ctx); err != nil {
+		return fmt.Errorf("UpdateRoot(): %v", err)
+	}
+	return c.getInclusionProof(ctx, leaf.MerkleLeafHash, c.root.TreeSize)
+}
+
+// VerifyInclusionAtIndex ensures that the given leaf data has been included in the log at a particular index.
```

Comment:
Please document that this causes a root update (as per UpdateRoot)

Btw, is that something we *need* to do? It's easier to reason about client state if it doesn't have implied calls.